macrosynergy/panel/view_ranges.py

```diff
@@ -47,19 +48,19 @@ def view_ranges(df: pd.DataFrame, xcats: List[str] = None,  cids: List[str] = No
     sns.set(style="darkgrid")
 
     if title is None:
-        if kind == 'bar':
-            title = f'Means and standard deviations from {s_date} to {e_date}'
-        elif kind == 'box':
-            title = f'Interquartile ranges, extended ranges and outliers ' \
-                    f'from {s_date} to {e_date}'
+        if kind == "bar":
+            title = f"Means and standard deviations from {s_date} to {e_date}."
+        elif kind == "box":
+            title = f"Interquartile ranges, extended ranges and outliers " \
+                    f"from {s_date} to {e_date}."
     if ylab is None:
         ylab = ""
 
     if sort_cids_by == 'mean':
-        dfx = df[df['xcat'] == xcats[0]].groupby(['cid'])[val].mean()
+        dfx = dfd.groupby(['cid'])[val].mean()
```

Comment:
Where does dfd come from? Also, what category's mean is the basis of the order?
The only solution seeems to be to order by first category, as before, but append any other cids that are not in order alphabetically,